src/data/nat/prime.lean

```diff
@@ -1127,16 +1146,19 @@ begin
   exact prime_of_mem_factors hqa
 end
 
-lemma factors_mul_of_coprime {a b : ℕ} (hab : coprime a b) (p:ℕ):
+lemma mem_factors_mul_of_coprime {a b : ℕ} (hab : coprime a b) (p : ℕ) :
   p ∈ (a * b).factors ↔ p ∈ a.factors ∪ b.factors :=
 begin
   rcases a.eq_zero_or_pos with rfl | ha,
   { simp [(coprime_zero_left _).mp hab] },
   rcases b.eq_zero_or_pos with rfl | hb,
   { simp [(coprime_zero_right _).mp hab] },
-  rw [mem_factors_mul_of_pos ha hb p, list.mem_union]
+  rw [mem_factors_mul_of_ne_zero ha.ne' hb.ne' p, list.mem_union]
 end
 
+lemma factors_mul_to_finset_of_coprime {a b : ℕ} (hab : coprime a b) :
```

Comment:
why have this? this is only better than the `ne_zero` version in the cases `0, 1`, and `1, 0`.

Reply:
We have a bunch of `of_coprime` versions of all the `pos` lemmas already, I'm just adding this one for consistency with those!

Reply:
I think this case is more common than expected, especially in applications - for instance to show that the number of prime factor function is multiplicative